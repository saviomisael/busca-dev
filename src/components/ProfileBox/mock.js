import Image from '../../assets/images/person_white.svg';

export const profileBoxPropsMock = {
  avatar: Image,
  username: 'johndoe',
  fullname: 'John Doe',
  bio: 'bio Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  blog: 'blog.com',
};
