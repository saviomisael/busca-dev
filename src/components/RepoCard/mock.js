export const repoCardPropsMock = {
  repoName: 'repo name',
  description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  starsCount: 0,
  repoUrl: 'github.com/repo',
};
