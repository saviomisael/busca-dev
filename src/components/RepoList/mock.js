export const repoListPropsMock = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
  ],
};

export const manyRepositories = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 2,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 3,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 3,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 4,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 5,
      repoUrl: 'github.com/repo',
    },
  ],
};
