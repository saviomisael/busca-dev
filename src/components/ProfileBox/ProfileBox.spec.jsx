import { screen } from '@testing-library/react';
import { ProfileBox } from '.';
import { profileBoxPropsMock } from './mock';

describe('<ProfileBox />', () => {
  it('should renders component with default props', () => {
    const props = { ...profileBoxPropsMock };

    delete props.bio;
    delete props.blog;

    global.renderTheme(<ProfileBox {...props} />);

    const bioElement = screen.queryByText(
      'bio Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
    );

    const blogElement = screen.queryByRole('link', { name: 'blog.com' });

    expect(bioElement).not.toBeInTheDocument();

    expect(blogElement).not.toBeInTheDocument();
  });

  it('should renders avatar image with alt containing username', () => {
    global.renderTheme(<ProfileBox {...profileBoxPropsMock} />);

    const avatarElement = screen.getByAltText(`${profileBoxPropsMock.username} avatar`);

    expect(avatarElement).toBeInTheDocument();
  });

  it('should renders bio when bio prop is truthy', () => {
    global.renderTheme(<ProfileBox {...profileBoxPropsMock} />);

    const bioElement = screen.getByText(/^bio Lorem ipsum/i);

    expect(bioElement).toBeInTheDocument();
  });

  it('should renders blog element when blog prop is truthy', () => {
    global.renderTheme(<ProfileBox {...profileBoxPropsMock} />);

    const blogElement = screen.getByRole('link', { name: 'blog.com' });

    expect(blogElement).toBeInTheDocument();
  });
});
