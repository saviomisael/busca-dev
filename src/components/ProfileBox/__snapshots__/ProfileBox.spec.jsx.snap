// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileBox /> should matches with snapshot 1`] = `
.c0 {
  padding: 1rem;
  text-align: center;
}

.c2 {
  text-align: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 1rem;
}

.c1 {
  width: 14.8rem;
  height: 14.8rem;
}

.c5 {
  color: #F5F5F5;
  font-size: 1.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 2rem;
  column-gap: 2rem;
}

.c5::before {
  content: url(link_white.svg);
  -webkit-transform: translateY(0.2rem);
  -ms-transform: translateY(0.2rem);
  transform: translateY(0.2rem);
}

.c3 {
  margin-top: 0;
  color: #F5F5F5;
  font-weight: 400;
  font-family: 'Ubuntu',sans-serif;
}

.c4 {
  font-size: 1.8rem;
  color: #F5F5F5;
}

<div
  class="c0"
>
  <img
    alt="johndoe avatar"
    class="c1"
    src="person_white.svg"
  />
  <div
    class="c2"
  >
    <h5
      class="c3"
    >
      @
      johndoe
    </h5>
    <p
      class="c4"
    >
      John Doe
    </p>
    <p
      class="c4"
    >
      bio Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
    <a
      class="c5"
      href="blog.com"
      target="_blank"
    >
      blog.com
    </a>
  </div>
</div>
`;
